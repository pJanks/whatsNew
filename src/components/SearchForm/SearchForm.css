input {
  margin-right: 20px;
  height: 40px;
  width: 280px;
}
