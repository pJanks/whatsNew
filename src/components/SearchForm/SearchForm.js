import React, { Component } from 'react';
import './SearchForm.css';

class SearchForm extends Component {
  constructor() {
    super();
    this.state = {
      query: ''
    }
  }

updateQuery = (event) => {
  this.setState({ query: event.target.value })
}

runSearch = (event) => {
  event.preventDefault();
  this.props.searchFromForm(this.state.query)
  this.setState({ query: '' })
}

enterPressed = (event) => {
    if (event.key === 'Enter') {
      this.updateQuery(event)
      this.runSearch(event)
    } else {
      return
    }
}

render = () => {
  return (<form>
    <button onClick={this.runSearch}>Search Here</button>
    <input
    type='text'
    onChange={this.updateQuery}
    value={this.state.query}
    onKeyPress={this.enterPressed}
    placeholder='Enter you search here'
    />
    </form>)
  }
}


export default SearchForm;
