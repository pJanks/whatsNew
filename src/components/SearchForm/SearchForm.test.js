import React from 'react';
import ReactDOM from 'react-dom';
import SearchForm from './SearchForm';
import { shallow } from 'enzyme';

describe('SearchForm', () => {
  let wrapper, instance;

  beforeEach(() => {
    wrapper = shallow(<SearchForm />)
    instance = wrapper.instance()
  });

  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot()
  });
});
