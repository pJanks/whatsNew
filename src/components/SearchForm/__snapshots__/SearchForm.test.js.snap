// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<form>
  <button
    onClick={[Function]}
  >
    Search Here
  </button>
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Enter you search here"
    type="text"
    value=""
  />
</form>
`;
